jsonargparse_tests/test_typehints.py

```diff
@@ -489,6 +492,42 @@ def test_mapping_nested_without_args(parser):
     assert {"b": {"c": 2}} == parser.parse_args(['--map={"b": {"c": 2}}']).map
 
 
+def test_typeddict_without_arg(parser):
+    if sys.version_info < (3, 8):
+        pytest.skip("TypedDict introduced in python 3.8")
+    parser.add_argument("--typeddict", type=TypedDict("MyDict", {}))
+    assert {} == parser.parse_args(["--typeddict={}"])["typeddict"]
+    pytest.raises(ArgumentError, lambda: parser.parse_args(['--typeddict={"a":1}']))
+    pytest.raises(ArgumentError, lambda: parser.parse_args(["--typeddict=1"]))
+
+
+def test_typeddict_with_args(parser):
+    if sys.version_info < (3, 8):
+        pytest.skip("TypedDict introduced in python 3.8")
+    parser.add_argument("--typeddict", type=TypedDict("MyDict", {"a": int}))
+    assert {"a": 1} == parser.parse_args(["--typeddict={'a': 1}"])["typeddict"]
+    # extra key
```

Comment:
Instead of comments, use `subtests` like done for example here
https://github.com/omni-us/jsonargparse/blob/af37f088f257a184f1f9e77812539d8a2e584276/jsonargparse_tests/test_typehints.py#L664